src/main/java/gregtech/api/recipes/RecipeMap.java

```diff
@@ -167,19 +166,38 @@ public void addRecipe(ValidationResult<Recipe> validationResult) {
                 return;
         }
         Recipe recipe = validationResult.getResult();
-        recipeList.add(recipe);
-
+        recipeTreeSet.add(recipe);
+
+        for (CountableIngredient countableIngredient : recipe.getInputs()) {
```

Comment:
You iterate over `recipe.getInputs()` here, does this include non-consumable items? From what I remember the non-consumable items were separate from just `inputs`.

Reply:
.notConsumable returns a input with item count of zero.
added a test thats show it works